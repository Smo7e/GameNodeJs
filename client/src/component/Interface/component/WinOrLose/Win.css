.image-win {
    position: absolute;
    width: 70%;
    height: 60%;
    background: url(../../image/1674289275_balthazar-club-p-bogataya-zhizn-estetika-oboi-4\ 1.png);
    top: 15%;
    left: 15%;
    z-index: 2;
    display: block;
  object-fit: cover;
}

.text-win {
    position: absolute;
    width: 1034px;
    height: 31px;
    font-family: 'Playfair Display';
    font-style: normal;
    font-weight: 30;
    font-size: 150px;
    line-height: 31px;
    letter-spacing: 0.26em;
    color: #00FF29;
    text-shadow: 3px 3px 0 #000, -3px -3px 0 #000, 3px -3px 0 #000, -3px 3px 0 #000;
    top: 30%;
    left: 24%;
    z-index: 2;
}

.text-win-1 {
    position: absolute;
    width: 1000px;
    height: 100px;
    font-family: 'Playfair Display';
    font-style: normal;
    font-weight: 800;
    font-size: 55px;
    line-height: 100px;
    text-align: center;
    letter-spacing: 0.05em;
    color:rgba(255, 229, 229, 1);
    text-shadow: 1.8px 1.5px 0 rgb(0, 0, 0), -1.8px -1.8px 0 rgb(0, 0, 0), 1.8px -1.8px 0 rgb(0, 0, 0);
    top: 45%;
    left: 17%;
    z-index: 2;
} 

.text-win-2 {
    position: absolute;
    width: 1000px;
    height: 100px;
    font-family: 'Playfair Display';
    font-style: normal;
    font-weight: 800;
    font-size: 50px;
    line-height: 10px;
    text-align: center;
    letter-spacing: 0.05em;
    color:rgba(255, 229, 229, 1);
    text-shadow: 1.8px 1.5px 0 rgb(0, 0, 0), -1.8px -1.8px 0 rgb(0, 0, 0), 1.8px -1.8px 0 rgb(0, 0, 0);
    top: 63%;
    left: 16%;
    z-index: 2;
} 