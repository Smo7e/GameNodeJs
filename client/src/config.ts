// UdSU
// const DOMAIN = 'http://studfront';
// const PORT = 81;

// dev
const DOMAIN = "http://localhost";
const PORT = 3001;

// prod
//...

export const HOST = PORT ? `${DOMAIN}:${PORT}` : `${DOMAIN}`;

export const MEDIATOR = {
    EVENTS: {
        SERVER_ERROR: "SERVER_ERROR",
        GET_MESSAGES: "GET_MESSAGES",
        GET_SCENE: "GET_SCENE",
        GET_USER: "GET_USER",
        GET_GAMERS: "GET_GAMERS",
        GET_FRIENDS: "GET_FRIENDS",
        GET_INVITES: "GET_INVITES",
        LOGIN: "LOGIN",
        SIGNUP: "SIGNUP",
        LOGOUT: "LOGOUT",
        GET_MOBS: "GET_MOBS",
        GET_ERROR: "GET_ERROR",
        PLAY_MUSIC: "PLAY_MUSIC", 
        STOP_MUSIC: "STOP_MUSIC", 
        SET_MUSIC_VOLUME: "SET_MUSIC_VOLUME", 
        UPDATE_ARR_BULLET_TRAJECTORY: "UPDATE_ARR_BULLET_TRAJECTORY",
    },
    TRIGGERS: {},
};
