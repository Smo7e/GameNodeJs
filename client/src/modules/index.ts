import { TError } from "./Server/types";
import Server from "./Server/Server";
import Mediator from "./Mediator/Mediator";
import Store from "./Store/Store";

export type { TError };
export { Server, Mediator, Store };
