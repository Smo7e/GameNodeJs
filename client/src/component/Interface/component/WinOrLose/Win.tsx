import React from "react";
import "./Win.css"
const Win: React.FC = () => {
    return <>
        <div className="image-win"></div>
            <div className="text-win">
            ПОБЕДА
        </div>
        <div className="text-win-1">
            Поздравляю, ты закрыл долги
        </div>
        <div className="text-win-2">
            Добро пожаловать в успешную жизнь
        </div>
    </>;
};
export default Win;