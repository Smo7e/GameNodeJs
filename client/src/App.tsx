import React, { useState } from "react";
import { HOST, MEDIATOR } from "./config";
import { Server, Mediator, Store, TError } from "./modules";
import SignUp from "./component/SignUp/SignUp";
import Game from "./modules/Game/Game";
import Menu from "./component/Menu/Menu";
import Login from "./component/Login/Login";
import Heroes from "./component/Heroes/Heroes";
import Interface from "./component/Interface/Interface";
import Lobby from "./component/Lobby/Lobby";
import ErrorMessage from "./modules/ErrorMessage/ErrorMessage";
import Parameters from "./component/Parameters/Parameters";
import Music from "./component/Parameters/Music";
import "./App.css";

export const ServerContext = React.createContext<Server>(null!);
export const MediatorContext = React.createContext<Mediator>(null!);
export const StoreContext = React.createContext<Store>(null!);

export enum EPAGES {
    SIGNUP,
    LOGIN,
    GAME,
    MENU,
    HEROES,
    LOBBY,
    PARAMETERS,
}

const MainApp = () => {
    const [epages, setEpages] = useState<EPAGES>(EPAGES.LOGIN);
    return (
        <>
            {epages === EPAGES.LOGIN ? (
                <Login epages={setEpages} />
            ) : epages === EPAGES.SIGNUP ? (
                <SignUp epages={setEpages} />
            ) : epages === EPAGES.MENU ? (
                <Menu epages={setEpages} />
            ) : epages === EPAGES.GAME ? (
                <>
                    <Game />
                    <Interface />
                </>
            ) : epages === EPAGES.HEROES ? (
                <Heroes epages={setEpages} />
            ) : epages === EPAGES.LOBBY ? (
                <Lobby epages={setEpages} />
            ) : epages === EPAGES.PARAMETERS ? (
                <Parameters epages={setEpages} />
            ) : (
                <></>
            )}
            <ErrorMessage />
            <Music />
        </>
    );
};

const App: React.FC = () => {
    const store = new Store();
    const mediator = new Mediator(MEDIATOR);
    const server = new Server(HOST, mediator, store);

    const { SERVER_ERROR } = mediator.getEventTypes();
    mediator.subscribe(SERVER_ERROR, (data: TError) => {
        console.log(data);
    });

    return (
        <StoreContext.Provider value={store}>
            <MediatorContext.Provider value={mediator}>
                <ServerContext.Provider value={server}>
                    <MainApp />
                </ServerContext.Provider>
            </MediatorContext.Provider>
        </StoreContext.Provider>
    );
};

export default App;
