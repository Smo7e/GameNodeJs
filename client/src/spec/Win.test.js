import { render, screen } from "@testing-library/react";
import Win from "../component/Interface/component/WinOrLose/Win";
import { MediatorContext, ServerContext } from "../App";
import { Mediator, Server, Store } from "../modules";
import { HOST, MEDIATOR } from "../config";

const store = new Store();
const mediator = new Mediator(MEDIATOR);
const server = new Server(HOST, mediator, store);
test("Win snapshot", () => {
    const { asFragment } = render(
        <MediatorContext.Provider value={mediator}>
            <ServerContext.Provider value={server}>
                <Win />
            </ServerContext.Provider>
        </MediatorContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
});
