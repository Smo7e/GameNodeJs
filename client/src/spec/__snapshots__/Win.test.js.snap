// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Win snapshot 1`] = `
<DocumentFragment>
  <div
    class="image-win"
  />
  <div
    class="text-win"
  >
    ПОБЕДА
  </div>
  <div
    class="text-win-1"
  >
    Поздравляю, ты закрыл долги
  </div>
  <div
    class="text-win-2"
  >
    Добро пожаловать в успешную жизнь
  </div>
</DocumentFragment>
`;
